ui/media/js/dnd.js

```diff
@@ -304,20 +312,27 @@ function restoreTaskToUI(task, fieldsToSkip) {
     if (!('use_upscale' in task.reqBody)) {
         useUpscalingField.checked = false
     }
-    if (!('mask' in task.reqBody)) {
+    if (!('mask' in task.reqBody) && maskSetting.checked) {
         maskSetting.checked = false
+        maskSetting.dispatchEvent(new Event("click"))
     }
     upscaleModelField.disabled = !useUpscalingField.checked
     upscaleAmountField.disabled = !useUpscalingField.checked
 
-    // Show the source picture if present
-    initImagePreview.src = (task.reqBody.init_image == undefined ? '' : task.reqBody.init_image)
-    if (IMAGE_REGEX.test(initImagePreview.src)) {
-        if (Boolean(task.reqBody.mask)) {
-            setTimeout(() => { // add a delay to insure this happens AFTER the main image loads (which reloads the inpainter)
+    // hide/show source picture as needed
+    if (IMAGE_REGEX.test(initImagePreview.src) && task.reqBody.init_image == undefined) {
+        // hide source image
+        initImageClearBtn.dispatchEvent(new Event("click"))
+    }
+    else if (task.reqBody.init_image !== undefined) {
+        // listen for inpainter loading event, which happens AFTER the main image loads (which reloads the inpainter)
+        document.addEventListener('imagePainterLoad', function() {
```

Comment:
Why not add a regular 'load' event listener? Like `initImagePreview.addEventListener('load', function..)`.

addEventListener() allows multiple functions to be enqueued for an event (that's why it's addEventListener, not setEventListener).

This way, we won't have to do a custom event dispatch in img2imgLoad().

Also, I don't think we need a custom 'load' dispatch in line 335. A 'load' event should be called automatically when the src image is loaded.